### PR TITLE
UserSessionProvider.removeUserSessions now removes all user sessions …

### LINKED ISSUE #31359: Offline sessions are not removed from admin console after sign out all active sessions
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

When "sign out all active sessions" are pushed in admin console, offline tokens are invalidated, but offline sessions are still displayed in "sessions". They should be removed from "sessions", because offline tokens are invalid.

### Version

25.0.1

### Regression

- [ ] The issue is a regression

### Expected behavior

Offline sessions in admin console "sessions" are removed when "sign out all active sessions" are pushed, and offline tokens are also invalidated.

### Actual behavior

When "sign out all active sessions" are pushed in admin console, offline tokens are invalidated, but offline sessions are still displayed in "sessions".

### How to Reproduce?

- Take access token and refresh token(offline token) with "offline_access" scope
- See "sessions" in admin console, and offline session is displayed
- Push "sign out all active sessions"
- Use offline token in token refresh flow. Response is "400 Bad Request: "{"error":"invalid_grant","error_description":"Stale token"}"{"error":"invalid_grant","error_description":"Stale token"}" and access token is not returned.
- See "sessions" in admin console, and offline session is still displayed

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by sguilhen**
…(both online and offline)

Closes #31359

### CODE REVIEW

File: testsuite/model/src/test/java/org/keycloak/testsuite/model/session/SessionTimeoutsTest.java [hunk 1]
```diff
@@ -93,9 +93,7 @@ public void cleanEnvironment(KeycloakSession s) {
         InfinispanTestUtil.revertTimeService(s);
         RealmModel realm = s.realms().getRealm(realmId);
         s.getContext().setRealm(realm);
-        UserModel user1 = s.users().getUserByUsername(realm, "user1");
         s.sessions().removeUserSessions(realm);
```
Comment (pedroigor) [inline testsuite/model/src/test/java/org/keycloak/testsuite/model/session/SessionTimeoutsTest.java:96]: Can we remove this line too because of the call to `removeRealm` below?

**Comment by ahus1** [comment]
IMHO this should be listed in the upgrading guide so people have a chance to learn about this changed behavior.

**Review by pedroigor — DISMISSED** [review]
@ahus1 @mposolda Would you mind also reviewing it?

A related, but follow-up, issue would be to change the session expiration job to consider invalidations using a `notBefore` timestamp. Today we only rely on the session lifetime and last refresh.

**Comment by sguilhen** [comment]
> IMHO this should be listed in the upgrading guide so people have a chance to learn about this changed behavior.

I'm not entirely sure we should move on with this - looking at the code, the general assumption of implementations for this method is that only regular sessions are to be removed.

We should perhaps keep it that way and introduce a `removeAllUserSessions` - problem is how to do that without breaking API compatibility. The provider has no equivalent method for removing all offline sessions, which prevents me from adding a default `removeAllUserSessions` impl that would simply call the existing `removeUserSessions` and the missing `removeOfflineUserSessions(RealmModel realm)`. 

Also, there's no method to retrieve all offline sessions of a realm (only by client or user), so I can't even have a naive default impl for this `removeAllUserSessions` that would call the `removeUserSessions` and then iterate over all offline sessions removing them one by one.

So bottom line is that I haven't found a good way to introduce a default method for removing all sessions without breaking API compatibility.

File: testsuite/model/src/test/java/org/keycloak/testsuite/model/session/SessionTimeoutsTest.java (hunk 1 above)
Comment (sguilhen) [reply testsuite/model/src/test/java/org/keycloak/testsuite/model/session/SessionTimeoutsTest.java:96]: Maybe the whole thing can just rely on the removeRealm() to clean everything up. The providers are fetching the session persister and calling `onRealmRemove` there and the JPA persister removes all sessions already.

**Comment by pedroigor** [comment]
> So bottom line is that I haven't found a good way to introduce a default method for removing all sessions without breaking API compatibility.

I don't think we need to worry about backward compatibility in this case. Effectively, sessions were invalidated already we are just making sure they are removed despite their type (offline vs online).

But still, a note in release notes make sense.

File: testsuite/model/src/test/java/org/keycloak/testsuite/model/session/SessionTimeoutsTest.java (hunk 1 above)
Comment (pedroigor) [reply testsuite/model/src/test/java/org/keycloak/testsuite/model/session/SessionTimeoutsTest.java:96]: Yeah, exactly.

**Comment by sguilhen** [comment]
I've sent another PR with a slightly different strategy - https://github.com/keycloak/keycloak/pull/34449

It should preserve the API compatibility and it also the semantics of the existing `removeUserSessions(RealmModel realm)` method in `UserSessionProvider`

**Review by pedroigor — DISMISSED** [review]

**Review by pedroigor — APPROVED** [review]
